### PR TITLE
Upgrade Ember 2.4

### PR DESCRIPTION
**Pull request by westlywright**
refactor to add some immutabilty and fix another bug

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -24,6 +24,7 @@
     "clipboard": "~1.5.5"
   },
   "resolutions": {
-    "ember": "2.3.1"
+    "ember": "2.3.1",
```
Comment (vincent99) [inline bower.json]: Should this be 2.4.x?

File: app/hosts/new/controller.js [hunk 2]
```diff
@@ -1,30 +1,20 @@
 import Ember from 'ember';
+import DriverChoices from 'ui/utils/driver-choices';
```
Comment (vincent99) [inline app/hosts/new/controller.js]: Was there a reason to move these?

File: bower.json (hunk 1 above)
Comment (westlywright) [inline bower.json]: I'll fix that

File: app/hosts/new/controller.js (hunk 2 above)
Comment (westlywright) [inline app/hosts/new/controller.js]: I left this in cause i'll be using it in my refactor of drivers. I was just decoupling my changes for the Driver changes but just thought i'd leave this in.

**Comment by vincent99** [comment]
@westlywright Can you separate out the bugs into a separate PR from the ember update.. I want to put the bugs into 1.0.1 but merge 2.4 in right after.

**Pull request merged**